File: qc-domain/src/main/java/com/eager/questioncloud/domain/review/model/QuestionReview.java
package com.eager.questioncloud.domain.review.model;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
public class QuestionReview {
    private Long id;
    private Long questionId;
    private Long reviewerId;
    private String comment;
    private int rate;
    private LocalDateTime createdAt;
    private Boolean isDeleted;

    @Builder
    public QuestionReview(Long id, Long questionId, Long reviewerId, String comment, int rate, LocalDateTime createdAt, Boolean isDeleted) {
        this.id = id;
        this.questionId = questionId;
        this.reviewerId = reviewerId;
        this.comment = comment;
        this.rate = rate;
        this.createdAt = createdAt;
        this.isDeleted = isDeleted;
    }

    public void modify(String comment, int rate) {
        this.comment = comment;
        this.rate = rate;
    }

    public void delete() {
        this.isDeleted = true;
    }

    public static QuestionReview create(Long questionId, Long reviewerId, String comment, int rate) {
        return QuestionReview.builder()
            .questionId(questionId)
            .reviewerId(reviewerId)
            .comment(comment)
            .rate(rate)
            .createdAt(LocalDateTime.now())
            .isDeleted(false)
            .build();
    }
}

File: qc-domain/qc-review/qc-review-core/src/main/kotlin/com/eager/questioncloud/review/domain/QuestionReview.kt
package com.eager.questioncloud.review.domain

import java.time.LocalDateTime

class QuestionReview(
    val id: Long = 0,
    val questionId: Long,
    val reviewerId: Long,
    var comment: String,
    var rate: Int,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    var isDeleted: Boolean = false,
) {
    fun modify(comment: String, newRate: Int): Int {
        val beforeRate = this.rate

        this.comment = comment
        this.rate = newRate

        return newRate - beforeRate
    }

    fun delete() {
        this.isDeleted = true
    }

    companion object {
        fun create(
            questionId: Long,
            reviewerId: Long,
            comment: String,
            rate: Int
        ): QuestionReview {
            return QuestionReview(
                questionId = questionId,
                reviewerId = reviewerId,
                comment = comment,
                rate = rate
            )
        }
    }
}
